Simplify standard issue templates

hillaryj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->
Removes existing unused issue templates and creates some more catch-all ones.

Updates each of the descriptions to provide a buffet of choices to funnel users into creating the right kind of issue.

## 💭 Motivation and Context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

Several issue types are unused, so we've tailored these to be more useful.

Closes https://github.com/cisagov/.github/issues/5.

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->

<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

Documentation-only changes; view the file to see the new setup.

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

hillaryj committed:
Update question template to direct to wiki and discussion boards

hillaryj committed:
Simplify templates

jsf9k commented on .github/ISSUE_TEMPLATE/bug.md:
```diff
@@ -1,27 +1,29 @@
+about: Describe a bug, error, unexpected behavior, or other problem
 ---
 
 # 🐛 Bug Report
```
```suggestion
# 🐛 Bug Report #
```

jsf9k commented on .github/ISSUE_TEMPLATE/bug.md:
```diff
@@ -1,27 +1,29 @@
-A clear and concise description of what the bug is.
+What's wrong? Please be specific.
 
 ## To Reproduce
```
```suggestion
## To Reproduce ##
```

hillaryj committed:
Remove trailing whitespace

jsf9k commented on .github/ISSUE_TEMPLATE/bug.md:
```diff
@@ -1,27 +1,29 @@
+1. Do this
+1. Then this
 
 ## Expected behavior
```
```suggestion
## Expected behavior ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/bug.md:
```diff
@@ -1,27 +1,29 @@
+What did you expect to happen that didn't?
 
-## Any helpful log output
+## Any helpful log output or screenshots
```
```suggestion
## Any helpful log output or screenshots ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+about: Describe a high-level objective which covers multiple issues
+---
+
+# 🚀 Summary
```
```suggestion
# 🚀 Summary #
```

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+An epic is a high-level objective encompassing multiple issues, as opposed to
+a specific unit of work.
+
+## Pitch
```
```suggestion
## Pitch ##
```

hillaryj committed:
Fix EOF

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+
+This would be useful because...
+
+## Implementation notes
```
```suggestion
## Implementation notes ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+- how this feature would act
+- any related work, especially links to issues
+
+## Acceptance Criteria
```
```suggestion
## Acceptance Criteria ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+about: Describe an idea, improvement, todo, feature, issue, story, or task
+---
+
+# 💡 Summary
```
```suggestion
# 💡 Summary #
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+
+What is the work, as a high-level summary?
+
+## Pitch
```
```suggestion
## Pitch ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+
+This would be useful because...
+
+## Implementation notes
```
```suggestion
## Implementation notes ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+- how this feature would act
+- any related work, especially links to issues
+
+## Acceptance Criteria
```
```suggestion
## Acceptance Criteria ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+
+Please provide details for implementation, such as:
+
+- an example for how this feature would be used
+- what this feature would look
+- how this feature would act
+- any related work, especially links to issues
```
I believe GitHub Markdown prefers `*` to `-`, although it accepts both.
```suggestion
* an example for how this feature would be used
* what this feature would look
* how this feature would act
* any related work, especially links to issues
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+- an example for how this feature would be used
+- what this feature would look
+- how this feature would act
+- any related work, especially links to issues
```
```suggestion
* an example for how this feature would be used
* what this feature would look
* how this feature would act
* any related work, especially links to issues
```

hillaryj replied:
I have a strong preference for dashes over asterisks for building a bulleted list in an issue, but don't have a definable reason other than "the asterisk lists bug my synesthesia."

I defer to the wisdom of the group on what to use.

hillaryj replied:
Our linter prefers/requires dashes, so I'll go with that to keep the build from failing.

jsf9k commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,14 +1,29 @@
+about: If you have questions, visit the Discussions and Wiki
 ---
 
 # 💬 Questions and Help
```
```suggestion
# 💬 Questions and Help #
```

jsf9k commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,14 +1,29 @@
 # 💬 Questions and Help
 
-## Please note that this issue tracker is not a help forum
+## For questions or help please visit other resources
```
```suggestion
## For questions or help please visit other resources ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,14 +1,29 @@
+  - [CyHy wiki](https://github.com/cisagov/cyhy-system/wiki)
+  - [CyHy discussion board](https://github.com/cisagov/cyhy-system/discussions)
 
 ## This issue will be closed
```
```suggestion
## This issue will be closed ##
```

hillaryj committed:
Sort keywords so specifics are first

jsf9k commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,14 +1,29 @@
+
+For specific `cisagov` systems, please see:
+
+- [COOL system](https://github.com/cisagov/cool-system/)
+  - [COOL wiki](https://github.com/cisagov/cool-system/wiki)
+  - [COOL discussion board](https://github.com/cisagov/cool-system/discussions)
+- [Cyber Hygiene (CyHy) system](https://github.com/cisagov/cyhy-system/)
+  - [CyHy wiki](https://github.com/cisagov/cyhy-system/wiki)
+  - [CyHy discussion board](https://github.com/cisagov/cyhy-system/discussions)
```
```suggestion
* [COOL system](https://github.com/cisagov/cool-system/)
  * [COOL wiki](https://github.com/cisagov/cool-system/wiki)
  * [COOL discussion board](https://github.com/cisagov/cool-system/discussions)
* [Cyber Hygiene (CyHy) system](https://github.com/cisagov/cyhy-system/)
  * [CyHy wiki](https://github.com/cisagov/cyhy-system/wiki)
  * [CyHy discussion board](https://github.com/cisagov/cyhy-system/discussions)
```

jsf9k reviewed (commented):
Some formatting suggestions.

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

jsf9k commented on .github/ISSUE_TEMPLATE/bug.md:
```diff
@@ -1,27 +1,29 @@
+What's wrong? Please be specific.
 
-## To Reproduce
+## To Reproduce ##
```
The capitalization for our secondary headers should be consistent.
```suggestion
## To reproduce ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+An epic is a high-level objective encompassing multiple issues, as opposed to
+a specific unit of work.
+
+## Pitch ##
+
+Why does this feature belong in this project?
+
+This would be useful because...
+
```
I'm not opposed to removing this section, although I can see it being useful sometimes.  Is there another name we could use for it?

hillaryj replied:
The Pitch section basically duplicated the Motivation section that existed before - if you like that name better, we could switch.

jsf9k replied:
Re "Pitch" vs "Motivation", I like the latter better.  "Pitch" is less precise in my mind, and makes me think of salesmen.

jsf9k replied:
What are you thoughts on the "Epic" and "Issue" templates being so similar?  What if we reserved the right to create epics and only allowed external users to create issues?

My concern is that folks who are not @cisagov/team-ois are not going to know whether they should create an epic or an issue, and the answer will always be to create an issue.  Why give them a confusing option that only makes sense for @cisagov/team-ois members?  Unless I'm mistaken, there is no click-o-matic way to convert a mis-typed epic into an issue; it's a manual copy-and-paste process.

I think we should either:
* Figure out how to _clearly_ distinguish between the epic and issue templates, or
* Drop the epic template as a template type

I assume we need epics in order to satisfy ZenHub.  Is there more than one way to identify epics for Zenhub?

jsf9k replied:
The "Pitch" vs "Motivation" part is resolved via commit d4bb29a, but I'd like to leave the "Epic" vs "Issue" open and hear from some other members of @cisagov/team-ois.

Also, is an epic issue type necessary for ZenHub?

dav3r replied:
As far as I can tell, the "epic" label is not required to exist in a GitHub repo in order for ZenHub to create epics in that repo (see [Intro to ZenHub Epics](https://help.zenhub.com/support/solutions/articles/43000010341-an-intro-to-zenhub-epics)).  Although I don't know if ZenHub will automatically create an "epic" label in a repo if necessary- that would require a little bit of testing.

The [Intro to ZenHub Epics](https://help.zenhub.com/support/solutions/articles/43000010341-an-intro-to-zenhub-epics) has some good verbiage that we could use to help explain the difference between an issue and an epic.

Also, it's worth noting that ZenHub has functionality that allows one to convert issues to epics and vice versa.

jsf9k replied:
> As far as I can tell, the "epic" label is not required to exist in a GitHub repo in order for ZenHub to create epics in that repo

I'm talking about the "epic" issue type here, not labels.

hillaryj replied:
They're such similar templates, I'd agree that it isn't necessary to duplicate. If someone wants to make an "epic" then they can just make a normal issue and tag it epic via labels or ZenHub.

dav3r replied:
Whoops, sorry about that.  In that case, I don't think ZenHub matters here.  I don't believe it cares about the GitHub issue type at all- it has it's own mechanism for determining if something is a ZenHub "issue" or "epic".

hillaryj replied:
Eliminated in https://github.com/cisagov/.github/pull/11/commits/b31c613af89e4a34c16dfe13f7321c7c61f7bbcc

jsf9k commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+- how this feature would act
+- any related work, especially links to issues
+
+## Acceptance Criteria ##
```
```suggestion
## Acceptance criteria ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+- how this would act
+- any related work, including links to related issues
+
+## Acceptance Criteria ##
```
```suggestion
## Acceptance criteria ##
```

jsf9k commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,29 @@
+
+What is the work, as a high-level summary?
+
+## Pitch ##
+
+Why does this work belong in this project?
+
+This would be useful because...
+
```
I'd like to remove this section, or rename it to something else.

jsf9k reviewed (commented):
More minor thangs, plus some hate directed at the "Pitch" sections.

Also, I'm troubled by the fact that the epic and issue templates are almost identical.  That would seem to indicate that they are the same type of thang, and maybe we should hide the difference from the end user.  Just a thought.

mcdonnnj commented on .github/ISSUE_TEMPLATE/bug.md:
````diff
@@ -1,27 +1,29 @@
 
 Paste the results here:
 
 ```bash
````
Since this is expected to be log-like output, and since we're already doing changes.
```suggestion
```console
```

mcdonnnj commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,32 @@
+---
+name: 🚀 Epic
```
Just a style suggestion, definitely a take-it-or-leave-it.

```suggestion
name: 📜 Epic
```

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

hillaryj committed:
Update titles and remove Pitch wording

hillaryj committed:
Update question.md

mcdonnnj commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,14 +1,29 @@
 ---
-name: 💬 Questions / Help
-about: If you have questions, please send us an Email
+name: 💬 Questions & Help
```
Just for visibility:
https://github.com/cisagov/.github/issues/5#issuecomment-750458857

hillaryj replied:
Addressed in https://github.com/cisagov/.github/pull/11/commits/37b275b4f5dbb951b3e408d54fa20e9e0adfa7c1

mcdonnnj reviewed (approved):
I didn't see any major hangups for me that weren't touched on by others. Thanks for taking this on to improve these.

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>
Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

hillaryj committed:
Combine epic type into regular issue

hillaryj committed:
Restore unordered list formatting

hillaryj committed:
Create QUESTIONS.md

Add our top repositories and some guidance on how to get to documentation or ask questions instead of using issues.

hillaryj committed:
Refer question issues to new QUESTIONS.md

jsf9k reviewed (approved):
This looks great to me.  Thank you for sifting through my zillion suggestions.

hillaryj commented:
I used the number of stars/forks of our repos to generate the list in QUESTIONS.md.

hillaryj committed:
Add Sparrow link

mcdonnnj commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -5,25 +5,10 @@ about: If you have questions, visit GitHub Discussions and Wiki pages
 
 # 💬 Questions #
```
I think that rather than having a `question.md` template, we should add a `config.yml` to modify the Issue Selector like so:

```yaml
---
# When we are comfortable with our templates, this should be changed to false
# to force new issues into an existing template.
blank_issues_enabled: true
contact_links:
  - name: Questions
    url: https://github.com/cisagov/.github/QUESTIONS.md
    about: Please also see our documentation and Discussion boards for common questions and topics.
```

@dav3r @jsf9k I would appreciate your thoughts as well. You can see a sample of what it would look like in one of my testing repos [here](https://github.com/mcdonnnj/github-actions-testing/issues/new/choose).

felddy replied:
I do this... take a look at:
https://github.com/felddy/foundryvtt-docker/blob/develop/.github/ISSUE_TEMPLATE/config.yml

jsf9k replied:
I like what @mcdonnnj and @felddy are suggesting.

dav3r replied:
I like the link to direct questions to the repo's GitHub Discussions, as @felddy has shown.

hillaryj replied:
Looks good! We can use that yml approach to link directly to Discussions in certain repositories and to QUESTIONS.md for the default/generic. Unless there's a way to have it relatively-refer to the current repository's Discussions.

hillaryj replied:
Implemented in https://github.com/cisagov/.github/pull/11/commits/402cf85cbbcc0a0f0073aa572aa4face1adb7e60

jsf9k replied:
> Looks good! We can use that yml approach to link directly to Discussions in certain repositories and to QUESTIONS.md for the default/generic. Unless there's a way to have it relatively-refer to the current repository's Discussions.

Would a relative link [like this](../issues) work?

jsf9k replied:
> > Looks good! We can use that yml approach to link directly to Discussions in certain repositories and to QUESTIONS.md for the default/generic. Unless there's a way to have it relatively-refer to the current repository's Discussions.
> 
> Would a relative link [like this](../issues) work?

Already merged :cry:

hillaryj replied:
The problem I see with that is that not all repositories have Discussions or Wiki pages existent or active. QUESTIONS.md will at least give someone a starting point for the common repositories if they wander in.

jsf9k replied:
OK

mcdonnnj replied:
> 
> 
> > Looks good! We can use that yml approach to link directly to Discussions in certain repositories and to QUESTIONS.md for the default/generic. Unless there's a way to have it relatively-refer to the current repository's Discussions.
> 
> Would a relative link [like this](../issues) work?

Remember that adding *anything* to `ISSUE_TEMPLATE` in a repository will break the automatic inclusion of these defaults. I had to manually add these issue templates to [cisagov/cyhy_amis](https://github.com/cisagov/cyhy_amis) in https://github.com/cisagov/cyhy_amis/pull/300 because of the monthly redeploy template we added.

I did not have success when testing a relative link in `config.yml` when I tried.

mcdonnnj reviewed (approved):
This is some strong work 💪. I did have one further piece of discussion regarding `question.md`, but it doesn't necessarily have to be resolved right now.

dav3r reviewed (approved):
This looks awesome to me - thanks @hillaryj for taking on this effort! 🎻

hillaryj committed:
Move questions to a config file link
